napari/utils/progress.py

```diff
@@ -75,7 +73,7 @@ def __init__(
 
         # get progress bar added to viewer
         try:
-            from .dialogs.activity_dialog import get_pbar
+            from .._qt.dialogs.activity_dialog import get_pbar
 
             pbar = get_pbar(self, nest_under=nest_under, **pbar_kwargs)
         except ImportError:
```

Comment:
I'd prefer avoiding the try..import at the module level and not in a function.  One approach may be to treat `utils.progress()` as a factory function returning a tdqm-like object.  This might look something like:

```python
# At the module level in utils.py
try:
   import .._qt.dialogs.activity_dialog import qet_pbar
   def progress(*args,**kwargs):
       # get_pbar() would have to be modified to return a tdqm-like
       pbar=get_pbar(*args,**kwargs) 
       return pbar if pbar else tdqm(*args,**kwargs)
except ImportError:
   def progress(*args,**kwargs):
       return tdqm(*args,**kwargs)
```

I think this might also simplify some of the conditionals in the progress class since the implementation could assume that qt was around etc.  The tdqm-like returned from `get_pbar` would be a version of this progress class.

Reply:
@DragaDoncila @tlambert03 thoughts/ preferences here? I don't mind. Would be nice to wrap this PR up though so we can merge

Reply:
@sofroniewn initially I liked this but after chatting with Talley a bit more we realized it's maybe going in the opposite direction to what we would like i.e. further coupling the `progress` model to the `qt` view. @tlambert03 had a different idea for how we could refactor `progress` to know nothing about `qt` - exploring that is on my todo list for today. I think for now we should merge this as is - I can always make @nclack's suggested changes in a new PR if we don't come up with a better way to separate them.

Reply:
ok sounds good